app/javascript/components/SearchQueryDisplay.js

```diff
@@ -36,19 +39,36 @@ function formatFacet(facet, index, numFacets) {
     </span>
   )
 }
+export const ClearAllButton = () => {
+  const selectionContext = useContext(SearchSelectionContext)
+  const searchContext = useContext(StudySearchContext)
 
-export default function SearchQueryDisplay({terms, facets}) {
+  const clearSearch = () => {
+    const defaultSearchParams = {
+      terms: '',
+      facets: {}
+    }
+    selectionContext.performSearch()
+    searchContext.updateSearch(defaultSearchParams)
+    selectionContext.updateSelection(defaultSearchParams)
```

Comment:
Sorry, I don't think I communicated this well when we were talking.  I want you to edit `SearchSelectionProvider.updateSelection` to take a `searchNow` argument, just like `updateFacet` does.  Then instead of having to have order-sensitive double-update-calls like this in each component, it can be handled in the selection provider.  Then these two lines will just become `selectionContext.updateSelection(defaultSearchParams, true)`

Reply:
yeah, like I said, I don't think we were understanding each other well yesterday. :-).  it wasn't a *necessary* change to get this working, but it'll make future components much better.